snuba/query/query.py

```diff
@@ -0,0 +1,130 @@
+from __future__ import annotations
+
+from deprecation import deprecated
+from typing import (
+    Any,
+    Iterable,
+    Mapping,
+    Optional,
+    Sequence,
+    Tuple,
+    TypeVar,
+    Union,
+)
+
+
+Condition = Union[
+    Tuple[Any, Any, Any],
+    Iterable[Any],
```

Comment:
I think you're going to want something stronger than `Iterable` here (probably `Sequence`), since you're going to generally want indexed access and the ability to iterate multiple times, neither of which `Iterable` guarantees.